schema/AttributeObject.md

```diff
@@ -8,27 +8,39 @@ and in the [Entitlement Object](EntitlementObject.md) to assert the attributes t
+| **Attribute Name** | `Blob` | Attribute Names are not globally unique. |
+| **Attribute Canonical Name** | `https://demo.com/attr/Blob` | Combination of `Attribute Namespace` and `Attribute Name`, separated by the string `/attr/`. Attribute Canonical Names are the _globally unique_ part of the attribute. |
+| **Attribute Value** | `Green` | Attribute Values are not globally unique. |
+| **Attribute Definition** | `{rule_type: AllOf, valid_values: [Green, Red, Purple]}` | Authorization-relevant metadata (rule type: AllOf/AnyOf/Hierarchy, allowed values, etc) associated with a specific, globally unique `Attribute Canonical Name`. Stored/managed by the authoritative owner of the attribute, separately from data or entity policy. |
+| **Attribute Instance** | `https://demo.com/attr/Blob/value/Green` | `Attribute Canonical Name` + (a specific value, valid as per the `Attribute Definition`) |
 
-If this check succeeds, the KAS permits a decrypt operation and returns a valid key which the client can decrypt and use to expose the file contents.
+> Key Point: Attribute Namespaces are not globally unique by themselves. Attribute Names are not globally unique by themselves. The combination of **both Namespace and Value** (the Canonical Name) _must_ be globally unique, and _must_ globally identify the Attribute.
 
-The AttributeObject alone does not define how the KAS will compare an entity's attribute to an object attribute when making an access decision.
-The KAS uses the namespaced object attributes in the [Policy Object](PolicyObject.md) to look up attribute policies from the cognizant authority
-to make its policy decisions. Clients writing policies should use best available information from their organizations to select which AttributeObjects to include to protect the policy.
+> Key Point: As Attribute Canonical Names are globally unique, and Attribute Definitions are associated with a specific Attribute Canonical Name, it follows that there can be _only one_ Attribute Definition globally, for a given Canonical Name.
 
-## Example
+> Key Point: Only an Attribute Instance (Canonical Name + Value) can used for authorization decisions, or added to [Data Policy Objects](PolicyObject.md) or [Entity Entitlement Objects](EntitlementObject.md)
+
+When encrypting, the client determines which Attribute Instances an entity must have in order to decrypt the payload and appends those Attribute Instance URIs to the data's [Data Policy Object](PolicyObject.md).
+
+When a decrypt is requested, the KAS checks the [Data Policy Object](PolicyObject.md) against the [Claims Object](ClaimsObject.md) from the requesting client to 
+ensure that the Attribute Instances that each entity "has" matches the Attribute Instances that the data "has", using the Attribute Definitions currently associated with each individual data Attribute to determine comparison rule (AnyOf/AllOf/Hierarchy).
```

Comment:
I think there is an extra space here.  Also, this description is consistent with the "has" terminology - but suggesting an alternative:
"ensure that the entity entitlements match the data attributes, using the Attribute Definitions currently associated with each individual data Attribute to determine comparison rule (AnyOf/AllOf/Hierarchy).

If this check succeeds, the KAS permits a decrypt operation and returns a valid key which the client can use to decrypt and expose the data contents."

Reply:
I revised this section a bit, thx.